### ISSUE TITLE
cljs.core/-, all arguments must be numbers in focus.cljc

### ISSUE THREAD

**Issue by daaku** [kkinnear/zprint#91]
In a project built with shadow-cljs I get this noise whenever my project is compiled:

```
------ WARNING #1 --------------------------------------------------------------
 File: zprint/focus.cljc:100:47
--------------------------------------------------------------------------------
  97 |         #_(println "range-ssv: start:" start "nwpath:" nwpath)
  98 |         start (skip-whitespace-ssv ssv start)
  99 |         ssv-next (next-ssv ssv start)
 100 |         end (if (and ssv-next (not= start 0)) (dec ssv-next) (dec (count ssv)))]
-----------------------------------------------------^--------------------------
 cljs.core/-, all arguments must be numbers, got [#{nil clj-nil} number] instead
--------------------------------------------------------------------------------
 101 |     #_(println "range-ssv:" [start end])
 102 |     [start end]))
--------------------------------------------------------------------------------
```

I'm using version `0.4.13`.

**Comment by kkinnear** [comment]
I find this very strange.  From what you are saying, this is a compile time problem -- which makes sense, as it _can't_ be a run-time problem -- the `(and ssv-next (not= start 0))` will ensure that the `(dec ssv-next)` will never be called if `ssv-next` is nil.  So, this looks like some kind of static analysis that is telling us that `ssv-next` could be nil.  Which it clearly can -- `nil` is a valid return from `next-ssv`.

I have managed to reproduce this.  This behavior appeared between Clojurescript 1.10.339 and 1.10.439.  I can reproduce it with figwheel as well as shadow-cljs.  

It is hard for me to believe:
1. That this is the only place in my code where a numeric function is protected from a `nil` value by being wrapped in an `if` function.
2. That this is actually supposed to be disallowed, or even warned about.

Indeed, the announcement for Clojurescript 1.10.439 mentions this issue [here](https://clojurescript.org/news/2018-11-02-release#_function_return_type_inference) as an improvement.  But it seems a bit aggressive to me in this case.  I suspect I can recast this code to work around this warning behavior.  But this is certainly not what I would expect to see.

Thank you for alerting me to this.

**Comment by daaku** [comment]
Does seem aggressive with the obvious `and` condition around it. Seems like the type inference isn't capable of narrowing assertions within the conditional branch.

It doesn't affect functionality, but it does create noise during compilation with the warning so a workaround would be much appreciated.

**Comment by kkinnear** [comment]
I fixed this in 0.4.14.  Posted a question on the clojurescript google group too, but no response there.
